src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const ACTIONS = {
+    ADD_PROPERTIES: 'addProperties',
+    REMOVE_PROPERTIES: 'removeProperties',
+    CLEAR: 'clear',
+  };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ACTIONS.ADD_PROPERTIES:
+        Object.assign(state, action.extraData);
+        break;
+      case ACTIONS.REMOVE_PROPERTIES:
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
+      case ACTIONS.CLEAR:
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
```

Comment:
Follow the checklist, please 
![image](https://github.com/mate-academy/js_stateful-object/assets/99403198/c8f00179-c162-4ce6-ab7c-083f2155d22e)